pkgs/applications/editors/emacs/default.nix

```diff
@@ -31,7 +32,8 @@ stdenv.mkDerivation rec {
     sha256 = "0cwgyiyymnx4xdg99dm2drfxcyhy2jmyf0rkr9fwj9mwwf77kwhr";
   };
 
-  patches = lib.optional stdenv.isDarwin ./at-fdcwd.patch;
+  patches = lib.optional stdenv.isDarwin ./at-fdcwd.patch
+    ++ lib.optional withCsrc ./install-C-src.patch;
```

Comment:
I was told about inline patches in some other PR - that they should be downloadable from external source. And now I think that this is indeed good policy - it pushes you to improving upstream build process and making it more universal =)